core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -25,32 +25,84 @@ oppia.constant(
 oppia.constant(
   'COLLECTION_RIGHTS_URL_TEMPLATE',
   '/collection_editor_handler/rights/<collection_id>');
+oppia.constant(
+  'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
+  '/explorationsummarieshandler/data?' +
+  'stringified_exp_ids=<stringified_exp_ids>&' +
+  'include_private_explorations=<include_private_explorations>');
 
 oppia.controller('CollectionEditor', ['$scope',
   'WritableCollectionBackendApiService', 'CollectionRightsBackendApiService',
-  'CollectionObjectFactory', 'SkillListObjectFactory',
+  'ExplorationSummaryBackendApiService', 'CollectionObjectFactory',
+  'SkillListObjectFactory', 'CollectionValidationService',
   'CollectionUpdateService', 'UndoRedoService', 'alertsService', function(
     $scope, WritableCollectionBackendApiService,
-    CollectionRightsBackendApiService, CollectionObjectFactory,
-    SkillListObjectFactory, CollectionUpdateService, UndoRedoService,
+    CollectionRightsBackendApiService, ExplorationSummaryBackendApiService,
+    CollectionObjectFactory, SkillListObjectFactory,
+    CollectionValidationService, CollectionUpdateService, UndoRedoService,
     alertsService) {
     $scope.collection = null;
     $scope.collectionId = GLOBALS.collectionId;
     $scope.collectionSkillList = SkillListObjectFactory.create([]);
     $scope.isPublic = GLOBALS.isPublic;
+    $scope.validationIssues = [];
+
+    var _validateCollection = function() {
+      if ($scope.isPublic) {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPublicCollection(
+            $scope.collection));
+      } else {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPrivateCollection(
+            $scope.collection));
+      }
+    };
+
+    var _updateCollection = function(newBackendCollectionObject) {
+      $scope.collection = CollectionObjectFactory.create(
+        newBackendCollectionObject);
+      $scope.collectionSkillList.setSkills(newBackendCollectionObject.skills);
+      _validateCollection();
+    };
 
     // Load the collection to be edited.
     WritableCollectionBackendApiService.fetchWritableCollection(
       $scope.collectionId).then(
-        function(collectionBackendObject) {
-          $scope.collection = CollectionObjectFactory.create(
-            collectionBackendObject);
-          $scope.collectionSkillList.setSkills(collectionBackendObject.skills);
-        }, function(error) {
+        _updateCollection, function(error) {
           alertsService.addWarning(
-            error || 'There was an error loading the collection.');
+            error || 'There was an error when loading the collection.');
         });
 
+    UndoRedoService.setOnChangedCallback(function(changeObject, wasApplied) {
```

Comment:
Ah, I just looked at the setOnChangeCallback doc. Hm. Maybe s/wasApplied/isForwardChange? Or something that conveys the notion of "going forward" -- I don't like "applied" because I think it applies to backward changes as well.